Consolidate updating guides

ptgott opened the pull request:
Include a how-to guide for the fast path we want most users to move to: using cloud-hosted Teleport Enterprise and enrolling agents in automatic updates. For other use cases and edge cases, include a general Updating Reference.

ptgott committed:
Consolidate updating guides

Include a how-to guide for the fast path we want most users to move to:
using cloud-hosted Teleport Enterprise and enrolling agents in automatic
updates. For other use cases and edge cases, include a general Updating
Reference.

hugoShaka commented on docs/pages/upgrading/automatic-agent-updates.mdx:
````diff
@@ -0,0 +1,207 @@
+...
+```
+
+Note that the `inventory ls` command will also list `teleport-auth` and
+`teleport-proxy` services. These services are managed by the Teleport team, and
+updates will be performed automatically.
````
The auth and proxy sevrices are only managed by us for cloud customers, self-hosted AU users are also responsible for updating their control plane services.

ptgott replied:
I had intended for this to be a guide for cloud-hosted clusters, but now realize that nothing in the introduction/prerequisites mentions this. I've added a prerequisite for a cloud-hosted cluster and a link to the reference for self-hosted users.

hugoShaka commented on docs/pages/upgrading/automatic-agent-updates.mdx:
````diff
@@ -0,0 +1,207 @@
+   $ tsh ssh "${USER?}@${HOST?}"
+   ```
+
+1. Ensure that the Teleport repository is properly configured to use the
+   `stable/cloud` channel, and install the `teleport-ent-updater` package. This
+   must be completed on each agent you would like to enroll into automatic
+   updates:
````
This channel is cloud-specific but it seems this guide is not

ptgott committed:
Respond to hugoShaka feedback

- Fix step numbering.
- Add cloud prerequisite and mention cloud-hosted clusters in the
  introduciton.

bernardjkim commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,213 @@
+each Teleport agent. The updater communicates with the Teleport Proxy Service to
+determine when an update is available. When an update is available, the updater
+will update the Teleport agent during the next maintenance window. However, if a
+critical update is availble, the Teleport agent will be updated outside the
```
```suggestion
critical update is available, the Teleport agent will be updated outside the
```

bernardjkim commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,213 @@
+`stable/v13`, those agents will require manual intervention to be upgraded to
+the next major version (adding a new APT/YUM/zypper repo for `stable/v14`).
+
+Alternatively, you can use the `stable/rolling` channel, which contains
```
I'm not really sure what the use case for `stable/rolling` was supposed to be, but for Teleport Cloud we instruct users to use the `stable/cloud` channel.

ptgott replied:
Thanks! The two paragraphs about release channels for older Teleport versions probably shouldn't be in this page. I've removed them, and also added a subsection to the reference about choosing a release channel, based on the text in `docs/pages/includes/repo-channels.mdx`.

bernardjkim commented on docs/pages/upgrading/automatic-agent-updates.mdx:
````diff
@@ -0,0 +1,213 @@
+   updates:
+
+   ```code
+   $ curl https://goteleport.com/static/install.sh | bash -s <Var name="version" /> cloud
````
Users need to ensure that they install/upgrade to a compatible version of teleport here, but there isn't really any mention of what version to use here.

bernardjkim commented:
I noticed that the static install script does not configure the `stable/cloud` channel for cloud installations. I've created a PR to update the script https://github.com/gravitational/next/pull/2488. We'll probably want to merge that first.

ptgott committed:
Respond to bernardjkim feedback

- Remove unintentionally preserved text about release channels for
  Teleport v14.
- Add instructions for determining the Teleport version to install.

bernardjkim commented:
We're trying to make automatic updates a requirement at this point.

Can we include more details about the Cloud updates process and use firmer language on requiring automatic updates? Something like I had in the previous PR?

```
Cloud-hosted Teleport clusters are updated on a weekly basis. Major version updates
are performed every 4 months. You can monitor and subscribe to the [Teleport Status](https://status.teleport.sh/)
page to be notified of scheduled updates.

To maintain version compatibility between your Teleport agents and the Teleport
Proxy Services, ensure that all your Teleport agents stay up to date with the Teleport
version of your Teleport Proxy Services. Failure to maintain version compatibility
for your Teleport agents may result in a degraded connection or disconnection of
those agents.
```

ptgott committed:
Add an intro paragraph with a recommendation

Urge users to enroll their agents in automatic upgrades per feedback
from bernardjkim.

ptgott commented:
> We're trying to make automatic updates a requirement at this point. We can't technically force automatic updates, but we should make it clear that their agents will lose compatibly if they don't keep their agents up to date with Teleport Cloud.
> 
> Can we include more details about the Cloud updates process and use firmer language on requiring automatic updates? Something like I had in the previous PR?

Added in [5c862d4](https://github.com/gravitational/teleport/pull/41160/commits/5c862d47ec1e712b88c1cee41271af383a718b2d)

ptgott commented:
@r0mant Would you have a moment to take a look at this one? Thanks!

ptgott committed:
Add check for the Enterprise teleport-kube-agent

Incorporates #41132

r0mant commented on docs/pages/upgrading.mdx:
```diff
@@ -1,47 +1,22 @@
-Since Teleport is a distributed system with a number of services that run on
-potentially many hosts, you should take care when upgrading the cluster to
-ensure that all components remain compatible.
+If you have a Teleport Enterprise (Cloud) account, you can [set up automatic
+Teleport agent updates](./upgrading/automatic-agent-updates.mdx) to ensure that
+the version of Teleport running on agents is always compatible with that of the
+Teleport cluster.
```
Can we make the language stronger here? Automatic upgrades is a requirement, not a suggestion.

r0mant commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,242 @@
+description: Describes how to set up automatic agent updates.
+---
+
+We strongly recommend that Teleport users set up automatic agent updates to
```
Similar to above, automatic upgrades is not a recommendation, it's a requirement.

r0mant commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,242 @@
+---
+title: Set up Automatic Agent Updates
+description: Describes how to set up automatic agent updates.
```
Since this is specifically for Cloud users, should we mention this in the title/description?

r0mant commented on docs/config.json:
```diff
@@ -29,29 +29,12 @@
-              "title": "Self-Hosted Kubernetes",
-              "slug": "/upgrading/self-hosted-kubernetes/",
-              "forScopes": ["enterprise", "oss"]
+                "title": "Set up Automatic Agent Updates",
```
This guide is for Cloud users it seems so I would call it out explicitly in the title.

ptgott replied:
I've made this guide for self-hosted users as well. In my original PR, the intention was to make Cloud the happy path, but I can see how that would be overly aggressive.

r0mant commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,242 @@
+`teleport-proxy` services. These services are managed by the Teleport team, and
+updates will be performed automatically.
+
+## Step 2/3. Enroll agents on Linux servers in automatic updates
```
It seems like this guide is targeting clusters with existing agents that aren't yet enrolled in auto-upgrades? We should make this clear I think. Cloud users enrolling new agents should get automatic upgrades enabled by default, is that correct?

ptgott replied:
Edited the prerequisites. Step 2 shows the user how to identify agents that are not enrolled in automatic updates.

r0mant commented on docs/pages/upgrading/automatic-agent-updates.mdx:
```diff
@@ -0,0 +1,242 @@
+Teleport agents are not updated immediately when a new version of Teleport is
+released, and agent updates can lag behind the cluster by a few days.
+
+If the Teleport agent has not been automatically updating for several weeks, you
+can consult the updater logs to help troubleshoot the problem.
```
Nit: Would be helpful to show how to "consult the logs". Maybe a `journalctl` command.

r0mant commented on docs/pages/upgrading/reference.mdx:
```diff
@@ -0,0 +1,502 @@
+tocDepth: 3
+---
+
+This guide explains how to upgrade components of a Teleport cluster in
+non-standard situations. 
```
I'm not sure why self-hosted users trying to setup auto-upgrade is a non-standard situation. Maybe rephrase this to say this is targeting self-hosted setups?

r0mant commented on docs/pages/upgrading/reference.mdx:
````diff
@@ -0,0 +1,502 @@
+...
+```
+
+## Self-hosted automatic updates
````
Why is this not a separate guide, peer to Cloud automatic upgrades? Seems weird to have this hidden inside reference.

r0mant commented on docs/pages/upgrading/reference.mdx:
```diff
@@ -0,0 +1,502 @@
+
+### Configuring the maintenance schedule
+
+To eanble automatic upgrades in your cluster, you must create a cluster
```
```suggestion
To enable automatic upgrades in your cluster, you must create a cluster
```

r0mant reviewed (commented):
Left some comments.

ptgott committed:
Respond to r0mant feedback

- Include self-hosted instructions in the how-to guide
- Include a journalctl command to check logs
- Edit the upgrading menu page and intro of the how-to guide to provide
  stronger language re: using automatic agent upgrades in cloud
  accounts.

ptgott committed:
Add `helm repo update` step

ptgott commented:
@r0mant Friendly ping. Thanks!

ptgott committed:
Fix linter errors

ptgott commented:
@r0mant Just checking if there's anything else we need to do before we can merge this simplification. Thanks!

ptgott committed:
Respond to r0mant feedback
